Sources/MongoSwift/MongoClient.swift

```diff
@@ -171,41 +171,20 @@ public class MongoClient {
         // Initialize mongoc. Repeated calls have no effect so this is safe to do every time.
         initializeMongoc()
 
-        var error = bson_error_t()
-        guard let uri = mongoc_uri_new_with_error(connectionString, &error) else {
-            throw parseMongocError(error)
-        }
-        defer { mongoc_uri_destroy(uri) }
-
-        // if retryWrites is specified, set it on the uri (libmongoc does not provide api for setting it on the client).
-        if let rw = options?.retryWrites {
-            mongoc_uri_set_option_as_bool(uri, MONGOC_URI_RETRYWRITES, rw)
-        }
+        // TODO: when we stop storing _client, we will store these options and use them to determine the return values
+        // for MongoClient.readConcern, etc.
+        var options = options ?? ClientOptions()
+        let connString = try ConnectionString(connectionString, options: &options)
 
-        self._client = mongoc_client_new_from_uri(uri)
+        self._client = mongoc_client_new_from_uri(connString._uri)
         guard self._client != nil else {
             throw UserError.invalidArgumentError(message: "libmongoc not built with TLS support")
         }
 
-        // if a readConcern is provided, set it on the client
-        if let rc = options?.readConcern {
-            mongoc_client_set_read_concern(self._client, rc._readConcern)
-        }
-
-        // if a readPreference is provided, set it on the client
-        if let rp = options?.readPreference {
-            mongoc_client_set_read_prefs(self._client, rp._readPreference)
-        }
-
-        // if a writeConcern is provided, set it on the client
-        if let wc = options?.writeConcern {
-            mongoc_client_set_write_concern(self._client, wc._writeConcern)
-        }
-
         self.encoder = BSONEncoder(options: options)
         self.decoder = BSONDecoder(options: options)
 
-        if options?.eventMonitoring == true { self.initializeMonitoring() }
+        if options.eventMonitoring == true { self.initializeMonitoring() }
```

Comment:
this `== true` can now be omitted.

Reply:
good catch, I'm surprised there isn't a compiler warning for this. fixed.